src/main/java/frc/robot/subsystems/drive/SwerveModules.java

```diff
@@ -0,0 +1,145 @@
+package frc.robot.subsystems.drive;
+
+import com.ctre.phoenix.sensors.CANCoder;
+import com.ctre.phoenix.sensors.CANCoderStatusFrame;
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.CANSparkMax.IdleMode;
+import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.kinematics.SwerveModuleState;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants.DriveConstants;
+
+
+public class SwerveModules extends SubsystemBase{
+
+    private final CANSparkMax driveMotor;
+    private final CANSparkMax rotationMotor;
+
+    private final PIDController drivePID = new PIDController(0, 0, 0);
+    private PIDController rotationPID = new PIDController(0,0,0);
+
+    //private final CANCoder driveEncoder;  it is built in
+    private final CANCoder rotationEncoder;
+
+    private final double initialOffsetRadians;
+
+    public SwerveModules(int driveModeID, int turningMotorID, int cancoderID, double measuredOffsetsRadians, boolean driveModeReversed, boolean turningMotorReversed 
```

Comment:
probably should rename "driveModeID" to driveMotorID

Reply:
changed